tests/foreman/cli/test_variables.py

```diff
@@ -0,0 +1,1120 @@
+        2.  Create a matcher with value that matches the default type.
+
+        @assert: Matcher is created for matched type.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_validate_matcher_non_existing_attribute(self):
+        """Test matcher creation for non-existing attribute.
+
+        @id: 27ef1ef0-1c89-47eb-89e0-3da161154513
+
+        @steps:
+
+        1.  Attempt to create a matcher with non existing attribute.
+
+        @assert: Matcher is not created for non-existing attribute.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @skip_if_bug_open('bugzilla', 1259174)
+    @stubbed()
+    @tier1
+    def test_positive_create_matcher(self):
+        """Create a Smart Variable with matcher.
+
+        @id: 42113584-d2db-4b91-8775-06bffee36be4
+
+        @steps:
+
+        1. Create a smart variable with valid name and default value.
+        2. Create a matcher for Host with valid value.
+
+        @assert:
+
+        1. The smart Variable with matcher is created successfully.
+        2. The variable is associated with host with match.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_create_matcher_attribute_priority(self):
+        """Matcher Value set on Attribute Priority for Host.
+
+        @id: 65144295-f0ca-4bd0-ae01-96c50ca829fe
+
+        @steps:
+
+        1.  Create variable with some default value.
+        2.  Set fqdn as top priority attribute.
+        3.  Create first matcher for fqdn with valid details.
+        4.  Create second matcher for some attribute with valid details.
+        Note - The fqdn/host should have this attribute.
+        5.  Go to YAML output of associated host.
+
+        @assert: The YAML output has the value only for fqdn matcher.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_create_matcher_attribute_priority(self):
+        """Matcher Value set on Attribute Priority for Host - alternate priority.
+
+        @id: 7e52b054-4fcb-4c58-ae49-0d3348d14570
+
+        @steps:
+
+        1.  Create variable with some default value.
+        2.  Set some attribute(other than fqdn) as top priority attribute.
+        Note - The fqdn/host should have this attribute.
+        3.  Create first matcher for fqdn with valid details.
+        4.  Create second matcher for attribute of step 3 with valid details.
+        5.  Go to YAML output of associated host.
+
+        @assert:
+
+        1.  The YAML output has the value only for step 5 matcher.
+        2.  The YAML output doesn't have value for fqdn/host matcher.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_create_matcher_merge_override(self):
```

Comment:
Could you please confirm that provided scenario here is really working through hammer?